ArduPlane/quadplane.cpp

```diff
@@ -637,7 +636,9 @@ bool QuadPlane::setup(void)
       that the objects don't affect the vehicle unless enabled and
       also saves memory when not in use
      */
-    motor_class = (enum AP_Motors::motor_frame_class)frame_class.get();
+    motor_class = (AP_Motors::motor_frame_class)frame_class.get();
```

Comment:
We have a new toy we should use here.

Where we define `frame_class, which should change it to read:

```
AP_Enum<AP_Motors::motor_frame_class> frame_class;
```
After than you don't need the .get() *or* the cast :-)

similarly for _type

Reply:
needed a cast to use them in switch statements
and casting to unsigned gets an "ambiguous conversion" compiler error in the call to ::printf, but not in the call to config_error???
--separate commit for this one